dashboard/client/src/components/DataOverviewTable.tsx

```diff
@@ -116,59 +121,146 @@ const DataOverviewTable = ({
             </TableRow>
           </TableHead>
           <TableBody>
-            {list.map(
-              ({
-                dataset,
-                state,
-                ray_data_current_bytes,
-                ray_data_output_bytes,
-                ray_data_spilled_bytes,
-                progress,
-                total,
-                start_time,
-                end_time,
-              }) => (
-                <TableRow key={dataset}>
-                  <TableCell align="center">{dataset}</TableCell>
-                  <TableCell align="center">
-                    <TaskProgressBar
-                      numFinished={progress}
-                      numRunning={
-                        state === "RUNNING" ? total - progress : undefined
-                      }
-                      numCancelled={
-                        state === "FAILED" ? total - progress : undefined
-                      }
-                      total={total}
-                    />
-                  </TableCell>
-                  <TableCell align="center">
-                    <StatusChip type="task" status={state} />
-                  </TableCell>
-                  <TableCell align="center">
-                    {memoryConverter(Number(ray_data_output_bytes.max))}
-                  </TableCell>
-                  <TableCell align="center">
-                    {memoryConverter(Number(ray_data_current_bytes.value))}/
-                    {memoryConverter(Number(ray_data_current_bytes.max))}
-                  </TableCell>
-                  <TableCell align="center">
-                    {memoryConverter(Number(ray_data_spilled_bytes.max))}
-                  </TableCell>
-                  <TableCell align="center">
-                    {formatDateFromTimeMs(start_time * 1000)}
-                  </TableCell>
-                  <TableCell align="center">
-                    {end_time && formatDateFromTimeMs(end_time * 1000)}
-                  </TableCell>
-                </TableRow>
-              ),
-            )}
+            {list.map((dataset, index) => (
+              <DatasetTable
+                datasetMetrics={dataset}
+                isExpanded={expanded[dataset.dataset as keyof {}]}
```

Comment:
instead of using `as` to override the type checking, let's make the type of `expanded` `Record<string, boolean>`. I think that will then allow any string be accepted as a key.